### PR TITLE
[TF FE] Provide single tensor names for inputs and outputs in SavedModel

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** A lot of confusion is around multiple names we use for input and output tensor names in generated IRs for SavedModel format. Users suffered from multiple names that can happen after fusion multiple Identity nodes. Also, OVMS complains on it. So we made decision to clean-up any tensor name for inputs and outputs except the main names, i.e. user specific ones.

**Ticket:** 109671

### CODE REVIEW

**Review by mvafin — APPROVED** [review]

File: src/frontends/tensorflow/src/translate_session.cpp [hunk 1]
```diff
@@ -39,37 +39,66 @@ std::vector<T> reorder_ops_by_names(const std::vector<std::string>& names, const
...
-    if (std::dynamic_pointer_cast<ov::opset8::Parameter>(node)) {
-        for (size_t i = 0; i < node->get_output_size(); ++i) {
-            const auto& node_names = node->get_output_tensor(i).get_names();
-            for (const auto& name : node_names) {
-                const auto& saved_model_name = saved_model_names->find(name);
-                if (saved_model_name != saved_model_names->end()) {
-                    set_node_name(saved_model_name->second, node);
-                    return true;
-                }
+// Adjust names of the tensor by mapping internal names to user specific ones using the model signature
```
Comment (gkrivor) [inline src/frontends/tensorflow/src/translate_session.cpp:42]: You lost docs for procedure

File: src/frontends/tensorflow/src/translate_session.cpp [hunk 2]
```diff
@@ -433,23 +462,24 @@ void TranslateSession::translate_graph(const ov::frontend::InputModel::Ptr& inpu
...
     }
 
-    if (saved_model_inputs.get() && saved_model_inputs->size() > 0) {
-        for (auto param : params) {
-            // If parameter isn't found in a map of known inputs - mark them as unused
-            // and try to remove it in the normalize step later
-            if (!apply_saved_model_names(param, saved_model_inputs)) {
-                param->get_output_tensor(0).add_names({"saved_model_unused"});
-            }
+    if (saved_model_inputs || saved_model_outputs) {
```
Comment (gkrivor) [inline src/frontends/tensorflow/src/translate_session.cpp:465]: I'm not sure I understand how you exclude MetaGraph.

Could you explain?

**Review by gkrivor — COMMENTED** [review]
Please, verify using full scope of tests before merge.

File: src/frontends/tensorflow/src/translate_session.cpp (hunk 2 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/translate_session.cpp:465]: All Metagraph models I observe do not have the signature and this functionality is not tested and was not covered before. Did you test it on your own?

**Comment by rkazants** [comment]
> Please, verify using full scope of tests before merge.

Already run

File: src/frontends/tensorflow/src/translate_session.cpp (hunk 2 above)
Comment (gkrivor) [reply src/frontends/tensorflow/src/translate_session.cpp:465]: As we discussed - no known models, and no proves it couldn't be made, in theory. So, I'm ok with original change.

**Review by gkrivor — APPROVED** [review]

**Pull request merged**